tsfresh/transformers/relevant_feature_augmenter.py

```diff
@@ -212,19 +212,24 @@ def transform(self, X):
 
         self.feature_extractor.set_timeseries_container(self.timeseries_container)
 
-        relevant_time_series_features = set(self.feature_selector.relevant_features) - set(pd.DataFrame(X).columns)
+        # We can only extract features that originate from time series
+        relevant_extraction_settings = FeatureExtractionSettings.from_columns(
+            list(set(self.feature_selector.relevant_features) - set(pd.DataFrame(X).columns)))
 
-        relevant_extraction_settings = FeatureExtractionSettings.from_columns(relevant_time_series_features)
         relevant_extraction_settings.set_default = False
         relevant_extraction_settings.IMPUTE = self.feature_extractor.settings.IMPUTE
-        relevant_feature_extractor = FeatureAugmenter(settings=relevant_extraction_settings,
-                                                      column_id=self.feature_extractor.column_id,
-                                                      column_sort=self.feature_extractor.column_sort,
-                                                      column_kind=self.feature_extractor.column_kind,
-                                                      column_value=self.feature_extractor.column_value)
 
-        relevant_feature_extractor.set_timeseries_container(self.feature_extractor.timeseries_container)
+        feature_augmenter_restricted = FeatureAugmenter(settings=relevant_extraction_settings,
+                                                        column_id=self.feature_extractor.column_id,
+                                                        column_sort=self.feature_extractor.column_sort,
+                                                        column_kind=self.feature_extractor.column_kind,
+                                                        column_value=self.feature_extractor.column_value)
 
-        X_augmented = relevant_feature_extractor.transform(X)
+        feature_augmenter_restricted.set_timeseries_container(self.feature_extractor.timeseries_container)
 
-        return X_augmented.copy().loc[:, self.feature_selector.relevant_features]
+        if self.evaluate_only_added_features:
+            X_tsfresh = feature_augmenter_restricted.transform(X).loc[:, self.feature_selector.relevant_features]
+            return pd.concat([X_tsfresh, X], axis=1)
```

Comment:
The functionality to add non-time series features does neither fit the `extract_features` (which just extract the time series features) nor the `select_features` method (which just filters all features, no madder if they originate from time series or are externally given).

In my opinion, the `relevant_feature_augmenter` is the right place for that functionality.

Regarding the complexity of the `relevant_feature_augmenter`, If you come up with a more elegant way, I am happy to hear about that.